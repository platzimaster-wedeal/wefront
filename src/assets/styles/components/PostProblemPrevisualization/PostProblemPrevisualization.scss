@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../Breakpoints.scss';

.post-problem-previsualization {
  position: relative;
  width: 100%;
  height: auto;

  &__info {
    display: flex;
    flex-flow: row nowrap;
    padding: $space-size-extra-small 0;
    border-bottom: 2px solid $color-black;

    h4 {
      color: $color-black-90;
      margin: 0;

      @include media(tablet) {
        @include font-size($font-size-medium);
      }
    }

    strong {
      color: $color-black-90;
      padding: 0 $space-size-small;
      @include font-size($font-size-small);
    }
  }

  &__description {
    padding: $space-size-small 0; 
    @include font-size($font-size-normal);

    @include media(tablet) {
      @include font-size($font-size-medium);
    }
  }
}