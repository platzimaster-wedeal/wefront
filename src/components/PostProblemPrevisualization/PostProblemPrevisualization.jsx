import React from "react";
// Styles
import "../../assets/styles/components/PostProblemPrevisualization/PostProblemPrevisualization.scss";

// Components
import PostProblemGraphicData from "../PostProblemGraphicData/PostProblemGraphicData";

const PostProblemPrevisualization = ({ title, idProblem, short_description, modality = 'Remote', location = 'COL' }) => {
 return (
  <div className="post-problem-previsualization">
   <div className="post-problem-previsualization__info">
    <h4>{title || "Error with my pc"}</h4>
    <strong>{idProblem || "#121515"}</strong>
   </div>
   <p className="post-problem-previsualization__description">
    {short_description || "I need some help with my problem"}
   </p>
   <PostProblemGraphicData location={location} modality={modality} />
  </div>
 );
};

export default PostProblemPrevisualization;
