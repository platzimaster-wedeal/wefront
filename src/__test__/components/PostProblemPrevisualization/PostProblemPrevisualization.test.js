import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import PostProblemPrevisualization from "../../../components/PostProblemPrevisualization/PostProblemPrevisualization";

describe("Test for the component PostProblemPrevisualization :D", () => {
 const PostProblemPrevisualizationMounted = mount(
  <PostProblemPrevisualization />
 );
 const PostProblemPrevisualizationCreated = create(
  <PostProblemPrevisualization />
 );

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(PostProblemPrevisualizationCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(PostProblemPrevisualizationMounted.length).toEqual(1);
  });
  test("Should the name of the problem be a string", () => {
   expect(
    typeof PostProblemPrevisualizationMounted.find(
     ".post-problem-previsualization__info h4"
    ).text()
   ).toEqual("string");
  });
  test("Should the description of the problem be a string", () => {
   expect(
    typeof PostProblemPrevisualizationMounted.find(
     ".post-problem-previsualization__description"
    ).text()
   ).toEqual("string");
  });
  test("Should the id of the problem be a number", () => {
   expect(
    typeof Number(
     PostProblemPrevisualizationMounted.find(
      ".post-problem-previsualization__info strong"
     )
      .text()
      .split("#")[1]
    )
   ).toEqual("number");
  });
 });
});
