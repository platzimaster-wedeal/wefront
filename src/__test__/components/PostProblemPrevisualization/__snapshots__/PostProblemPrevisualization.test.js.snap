// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component PostProblemPrevisualization :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="post-problem-previsualization"
>
  <div
    className="post-problem-previsualization__info"
  >
    <h4>
      Error with my pc
    </h4>
    <strong>
      #121515
    </strong>
  </div>
  <p
    className="post-problem-previsualization__description"
  >
    I need some help with my problem
  </p>
  <div
    className="post-problem-graphic-data"
  >
    <div
      className="graphic-data graphic-data--active"
    >
      <svg
        fill="currentColor"
        height={30}
        size={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
      <strong>
        COL
      </strong>
    </div>
    <div
      className="graphic-data undefined"
    >
      <svg
        fill="currentColor"
        height={30}
        size={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
        />
      </svg>
      <strong>
        Day
      </strong>
    </div>
    <div
      className="graphic-data undefined"
    >
      <svg
        fill="currentColor"
        height={30}
        size={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 11c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 2c0-3.31-2.69-6-6-6s-6 2.69-6 6c0 2.22 1.21 4.15 3 5.19l1-1.74c-1.19-.7-2-1.97-2-3.45 0-2.21 1.79-4 4-4s4 1.79 4 4c0 1.48-.81 2.75-2 3.45l1 1.74c1.79-1.04 3-2.97 3-5.19zM12 3C6.48 3 2 7.48 2 13c0 3.7 2.01 6.92 4.99 8.65l1-1.73C5.61 18.53 4 15.96 4 13c0-4.42 3.58-8 8-8s8 3.58 8 8c0 2.96-1.61 5.53-4 6.92l1 1.73c2.99-1.73 5-4.95 5-8.65 0-5.52-4.48-10-10-10z"
        />
      </svg>
      <strong>
        Remote
      </strong>
    </div>
  </div>
</div>
`;
